src/test/PoolSwapTest.sol

```diff
@@ -106,13 +106,17 @@ contract PoolSwapTest is PoolTestBase {
 
         if (deltaAfter0 < 0) {
             if (data.testSettings.currencyAlreadySent) {
+                // Todo: This is a downside of sync!!! We must call sync before transferring in assets.
```

Comment:
maybe we should just get rid of these test cases? or add sync before them like we would do if we were swap V2->V4